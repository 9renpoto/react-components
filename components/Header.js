"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const classnames = require("classnames");
const React = require("react");
const NAVBAR = 'navbar';
const ID = `navMenu`;
class Header extends React.PureComponent {
    constructor(props) {
        super(props);
        this.state = {
            isActive: false
        };
    }
    render() {
        const { Logo, titles, baseUrl } = this.props;
        return (React.createElement("header", { className: `${NAVBAR}` },
            React.createElement("div", { className: `${NAVBAR}-brand` },
                Logo(`${NAVBAR}-item`),
                React.createElement("div", { className: classnames({
                        [`${NAVBAR}-burger burger`]: true,
                        'is-active': this.state.isActive
                    }), "data-target": ID, onClick: _ => this.setState({ isActive: !this.state.isActive }) },
                    React.createElement("span", null),
                    React.createElement("span", null),
                    React.createElement("span", null))),
            React.createElement("div", { className: classnames({
                    [`${NAVBAR}-menu`]: true,
                    'is-active': this.state.isActive
                }), id: ID },
                React.createElement("div", { className: `${NAVBAR}-end` }, titles.map((title, index) => (React.createElement("a", { className: `${NAVBAR}-item`, href: `${baseUrl}/${title}/`, key: index }, title)))))));
    }
}
exports.Header = Header;
//# sourceMappingURL=Header.js.map