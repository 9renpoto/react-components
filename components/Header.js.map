{"version":3,"file":"Header.js","sourceRoot":"","sources":["../../src/components/Header.tsx"],"names":[],"mappings":";;AAAA,yCAAwC;AACxC,+BAA8B;AAY9B,MAAM,MAAM,GAAG,QAAQ,CAAA;AACvB,MAAM,EAAE,GAAG,SAAS,CAAA;AAEpB,YAAoB,SAAQ,KAAK,CAAC,aAA2B;IAC3D,YAAa,KAAY;QACvB,KAAK,CAAC,KAAK,CAAC,CAAA;QACZ,IAAI,CAAC,KAAK,GAAG;YACX,QAAQ,EAAE,KAAK;SAChB,CAAA;IACH,CAAC;IACD,MAAM;QACJ,MAAM,EAAE,IAAI,EAAE,MAAM,EAAE,OAAO,EAAE,GAAG,IAAI,CAAC,KAAK,CAAA;QAC5C,MAAM,CAAC,CACL,gCAAQ,SAAS,EAAE,GAAG,MAAM,EAAE;YAC5B,6BAAK,SAAS,EAAE,GAAG,MAAM,QAAQ;gBAC9B,IAAI,CAAC,GAAG,MAAM,OAAO,CAAC;gBACvB,6BACE,SAAS,EAAE,UAAU,CAAC;wBACpB,CAAC,GAAG,MAAM,gBAAgB,CAAC,EAAE,IAAI;wBACjC,WAAW,EAAE,IAAI,CAAC,KAAK,CAAC,QAAQ;qBACjC,CAAC,iBACW,EAAE,EACf,OAAO,EAAE,CAAC,CAAC,EAAE,CAAC,IAAI,CAAC,QAAQ,CAAC,EAAE,QAAQ,EAAE,CAAC,IAAI,CAAC,KAAK,CAAC,QAAQ,EAAE,CAAC;oBAE/D,iCAAQ;oBACR,iCAAQ;oBACR,iCAAQ,CACJ,CACF;YACN,6BACE,SAAS,EAAE,UAAU,CAAC;oBACpB,CAAC,GAAG,MAAM,OAAO,CAAC,EAAE,IAAI;oBACxB,WAAW,EAAE,IAAI,CAAC,KAAK,CAAC,QAAQ;iBACjC,CAAC,EACF,EAAE,EAAE,EAAE;gBAEN,6BAAK,SAAS,EAAE,GAAG,MAAM,MAAM,IAC5B,MAAM,CAAC,GAAG,CAAC,CAAC,KAAK,EAAE,KAAK,EAAE,EAAE,CAAC,CAC5B,2BACE,SAAS,EAAE,GAAG,MAAM,OAAO,EAC3B,IAAI,EAAE,GAAG,OAAO,IAAI,KAAK,GAAG,EAC5B,GAAG,EAAE,KAAK,IAET,KAAK,CACJ,CACL,CAAC,CACE,CACF,CACC,CACV,CAAA;IACH,CAAC;CACF;AAhDD,wBAgDC"}