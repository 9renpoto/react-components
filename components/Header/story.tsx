import { storiesOf } from '@storybook/react'
import { random } from 'faker'
import * as React from 'react'

import { Header } from '.'

export function DefaultHeader () {
  return (
    <Header
      Logo={(className: string) => (
        <a href='#' className={className}>
          {random.word()}
        </a>
      )}
      titles={['blog', 'profile', 'slides']}
      baseUrl='#'
    />
  )
}

storiesOf('Content', module)
  .add('Header', () => <DefaultHeader />)
  .add('Content', () => (
    <main>
      <DefaultHeader />
      <section className='hero is-primary'>
        <div className='hero-body'>
          <div className='container'>
            <h1 className='title'>Hero title</h1>
            <h2 className='subtitle'>Hero subtitle</h2>
          </div>
        </div>
      </section>
    </main>
  ))
