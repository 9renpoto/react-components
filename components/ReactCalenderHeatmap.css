.react-calendar-heatmap {
  & text {
    font-size: 10px;
    fill: #aaa;
  }

  & [rect:hover] {
    stroke: #555;
    stroke-width: 1px;
  }

  & .color-github {
    &-0 {
      fill: #eee;
    }

    &-1 {
      fill: #d6e685;
    }

    &-2 {
      fill: #8cc665;
    }

    &-3 {
      fill: #44a340;
    }

    &-4 {
      fill: #1e6823;
    }
  }
}
