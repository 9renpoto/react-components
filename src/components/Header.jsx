/* @flow */
import React, { Component } from 'react'

type Props = {
  Logo: (className: string) => React$Element<any>,
  titles: Array<string>,
  baseUrl: string
}

export class Header extends Component<Props> {
  render () {
    const { Logo, titles, baseUrl } = this.props
    const logoClassName = 'nav-item title'
    return (
      <header className='nav'>
        <div className='nav-left'>
          {Logo(logoClassName)}
        </div>
        <div className='nav-right nav-menu'>
          {titles.map((title, index) =>
            <a className='nav-item' href={`${baseUrl}/${title}/`} key={index}>
              {title}
            </a>
          )}
        </div>
      </header>
    )
  }
}
