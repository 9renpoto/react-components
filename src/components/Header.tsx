import * as React from 'react'

export interface Props {
  Logo: Function
  titles: Array<string>
  baseUrl: string
}

export class Header extends React.Component<Props, {}> {
  render () {
    const { Logo, titles, baseUrl } = this.props

    const items = titles.map((title, index) => (
      <a className='nav-item' href={`${baseUrl}/${title}/`} key={index}>
        {title}
      </a>
    ))

    return (
      <header className='nav'>
        <div className='nav-left'>{Logo('nav-item title')}</div>
        <div className='nav-right nav-menu'>{items}</div>
      </header>
    )
  }
}
