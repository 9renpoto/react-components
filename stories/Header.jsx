/* @flow */
import React from 'react'
import { storiesOf } from '@storybook/react'
import { random } from 'faker'

import { Header } from '../src'

storiesOf('Content', module).add('Header', () =>
  <Header
    title={random.word()}
    titles={['blog', 'profile', 'slides']}
    baseUrl='#'
  />
)
