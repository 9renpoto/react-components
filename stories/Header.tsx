import { storiesOf } from '@storybook/react'
import { random } from 'faker'
import * as React from 'react'

import { Header } from '../src'

storiesOf('Content', module).add('Header', () => (
  <Header
    Logo={(className: string) => (
      <a href='#' className={className}>
        {random.word()}
      </a>
    )}
    titles={['blog', 'profile', 'slides']}
    baseUrl='#'
  />
))
