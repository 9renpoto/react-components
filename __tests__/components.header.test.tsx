/* @flow */
import { create } from 'react-test-renderer'

import * as React from 'react'
import { Header } from '../src/components/Header'

describe('header', () => {
  test('render', () => {
    function foo (name: string) {
      return (
        <div className={name}>
          {'Hello'}
        </div>
      )
    }

    const wrapper = create(
      <Header baseUrl={'http://example.com'} Logo={foo} titles={['t1', 't2']} />
    )
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
