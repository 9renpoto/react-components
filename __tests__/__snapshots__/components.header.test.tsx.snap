// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header render 1`] = `
<header
  className="navbar"
>
  <div
    className="navbar-brand"
  >
    <div
      className="navbar-item"
    >
      Hello
    </div>
    <div
      className="navbar-burger burger"
      data-target="navMenu"
      onClick={[Function]}
    >
      <span />
      <span />
      <span />
    </div>
  </div>
  <div
    className="navbar-menu"
    id="navMenu"
  >
    <div
      className="navbar-end"
    >
      <a
        className="navbar-item"
        href="http://example.com/t1/"
      >
        t1
      </a>
      <a
        className="navbar-item"
        href="http://example.com/t2/"
      >
        t2
      </a>
    </div>
  </div>
</header>
`;
